src/test/suite/completion.test.ts

```diff
@@ -44,4 +61,53 @@ describe("Should do completion", () => {
 
     expect(completions?.items).to.shallowDeepEqual(aCompletionResult());
   });
+  it("should accept an inline completion", async () => {
+    await isProcessReadyForTest();
+    mockAutocomplete(
+      requestResponseItems,
+      anAutocompleteResponse(INLINE_PREFIX, INLINE_NEW_PREFIX)
+    );
+    await moveToActivePosition();
+    await makeAChange(SINGLE_CHANGE_CHARACTER);
+    await triggerInline();
+
+    await sleep(1000);
+
+    await acceptInline();
+
+    expect(vscode.window.activeTextEditor?.document.getText()).to.equal(
+      INLINE_NEW_PREFIX
+    );
+  });
+  it("should prefer the popup when only popup is visible and there is no inline suggestion", async () => {
+    await assertSuggestionWith("console");
+  });
+  it("should prefer an inline when both popup and inline are visible", async () => {
+    await assertSuggestionWith("console.log", () => {
+      mockAutocomplete(
+        requestResponseItems,
+        anAutocompleteResponse("console", "console.log")
+      );
+    });
+  });
```

Comment:
its not clear what are these tests actually do. 
along with my other comment, i'd recommend extracting the logic to more functions